sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -590,6 +590,39 @@ def test_one_hot_encoder_feature_names_unicode():
     assert_array_equal(['n👍me_c❤t1', 'n👍me_dat2'], feature_names)
 
 
+@pytest.mark.parametrize("drop", ['first',
+                                  ['c', 2, 'b']],
+                         ids=['first', 'manual'])
+def test_one_hot_encoder_feature_names_drop(drop):
+    X = [['c', 2, 'a'],
+         ['b', 2, 'b']]
+
+    # Assume "drop=None" version has the right names in the right order
+    ohe_base = OneHotEncoder()
+    ohe_base.fit(X)
+
+    if drop == 'first':
+        drop_cats = [x[0] for x in ohe_base.categories_]
+    else:
+        drop_cats = drop
+
+    expected_names = list(ohe_base.get_feature_names())
+    for i, cat_drop in enumerate(drop_cats):
+        feat_drop = "x{}_{}".format(i, cat_drop)
+        expected_names.remove(feat_drop)
```

Comment:
This is a little bit involved. Seems like a good way to construct the expected_names without just reimplementing it. It's also robust to any changes in basic ordering.

Reply:
I don't think you benefit from this logic. Simpler (and more obviously correct) just to have the expected list of feature names as input to the function.

Reply:
Fair enough.